ereuse_devicehub/resources/action/views.py

```diff
@@ -158,13 +203,47 @@ def snapshot(self, snapshot_json: dict, resource_def):
         # Check if HID is null and add Severity:Warning to Snapshot
         if snapshot.device.hid is None:
             snapshot.severity = Severity.Warning
+
         db.session.add(snapshot)
         db.session().final_flush()
         ret = self.schema.jsonify(snapshot)  # transform it back
         ret.status_code = 201
         db.session.commit()
         return ret
 
+    def live(self, snapshot):
+        """If the device.allocated == True, then this snapshot create an action live."""
+        device = snapshot.get('device')  # type: Computer
+        # TODO @cayop dependency of pulls 85 and 83
+        # if the pr/85 and pr/83 is merged, then you need change this way for get the device
+        if not device.hid or not Device.query.filter(Device.hid==device.hid).count():
+            return
+
+        device = Device.query.filter(Device.hid==device.hid).one()
+
+        if not device.allocated:
+            return
+
+        time = 0
+        serial_number = ''
+        for hd in snapshot['components']:
```

Comment:
I suppose that we are not contemplating the case at the moment, but if there are multiple DataStorage with the SMART and the power_cycle_count variable, at this point we will have problems to take the live correctly (should the times be added up? I think not, but I would leave it for future discussion)

Reply:
When one live action is saved we save too the serial_number of the disk. When arrive an other live this search for this disk. Then always we compare with the same disk. If the device have 2 disks I suppose than work more or less the same time. Than only  we need the data liftime of one disk